rasa/graph_components/validators/finetuning_validator.py

```diff
@@ -230,32 +231,19 @@ def _get_fingerprint_of_domain_without_responses(domain: Domain) -> Text:
         domain.responses = {}
         return domain.fingerprint()
 
-    @staticmethod
-    def _get_fingerprint_of_config_without_epochs(
-        config: Optional[Dict[Text, Any]],
-    ) -> Text:
-        """Returns a fingerprint of the given configurations with "epoch" keys removed.
-
-        Note that the epoch keys are only removed if they are sub-keys of a
-        "pipeline" or "policies" key.
+    def _get_fingerprint_of_configs_without_epochs(self,) -> Text:
+        """Returns a fingerprint of the given configuration with "epoch" keys removed.
 
-        Args:
-            config: a configuration
         Returns:
             fingerprint
         """
-        if not config:
-            return ""
-
-        copied_config = copy.deepcopy(config)
-
-        for key in ["pipeline", "policies"]:
-            if copied_config.get(key):
-                for p in copied_config[key]:
-                    if "epochs" in p:
-                        del p["epochs"]
-
-        return rasa.shared.utils.io.deep_container_fingerprint(copied_config)
+        config_copies = [
+            copy.deepcopy(schema_node.config)
```

Comment:
thanks for the `deepcopy` 😆 💯

Reply:
I think we also need to evaluate the overall structure of the graph, don't we? E.g. things like changed order. How about also including the `needs` section in the fingerprint?

Reply:
And I think we need a change [here](https://github.com/RasaHQ/rasa/blob/f5b2ff4d208f5d376bd65c38eb33f8deb3a67b6a/rasa/engine/training/graph_trainer.py#L110) as well to put the `train_schema` in the execution context instead of the fingerprint schema (the validation will run during the fingerprint run - currently it would get the wrong schema)

Reply:
>I think we also need to evaluate the overall structure of the graph, don't we? 

how could I overlook this 😱  - think if we want to catch everything, we also need the "uses" part

Reply:
>  to put the train_schema in the execution context instead of the fingerprint schema (the validation will run during the fingerprint run - currently it would get the wrong schema)

will that break something else if we just replace it? 🤔  will look into it later

Reply:
It shouldn't as nothing is really using the graph schema in the execution context 🤔